### ISSUE TITLE
Discrepancy in Results Between W&B Tuning and Local Workflow

### ISSUE THREAD

**Issue by fy711** [MICS-Lab/scyan#39]
## Description
I've encountered an issue while trying to reproduce hyperparameter tuning results from Weights & Biases in my local workflow using the Scyan package.

## Steps to Reproduce
1. Performed hyperparameter tuning using Hydra from the terminal with the following command:

   ```
   python -m scripts.run -m project=my_project trainer=mps wandb.mode=online wandb.save_umap=true save_predictions=true
   ```

2. Selected a specific set of hyperparameters based on W&B results.
3. Attempted to replicate these results in my main workflow using the following code:

   ```python
   adata, table = scyan.data.load("my_project")
   model = scyan.Scyan(adata, table, **model_hyperparameters_from_wb)
   model.fit()
   model.predict()
   ```

## Expected Behavior
The prediction results in my main workflow should match those observed in Weights & Biases.

## Actual Behavior
The prediction results in my main workflow vary from what I see in Weights & Biases.

## Environment
- Scyan version: 1.6.2 (local pip install dev mode)
- Python version: 3.10
- Operating System: macOS-14.6-arm64

## Questions
1. Is the discrepancy with reproducing W&B results in local workflows expected or is it a issue?
2. Are there any additional steps I should take to ensure consistency between W&B and local runs?
3. How can I access the prediction results created by hydra?
4. Are there any updates on using the MPS trainer? Currently I'm using the fallback option but it will be fantastic if I can utilize GPU.

Thank you very much for your help and for developing this wonderful package!

**Comment by quentinblampey** [comment]
Hi @fy711,
Are the results very different with or without W&B? I think it may be simply due to the random seed. In the W&B scripts, we change the seed at each run (see [here](https://github.com/MICS-Lab/scyan/blob/583d356ef6eb73e84e07551004622f47d3c3b599/scripts/run.py#L37)). Can you try to set it at 0 at each step, and see if the results are the same?

By default, the predictions are not saved when using W&B. You can do so by setting `save_predictions` to `true` in the config [here](https://github.com/MICS-Lab/scyan/blob/583d356ef6eb73e84e07551004622f47d3c3b599/config/config.yaml#L15), and the results should be stored as a CSV in the hydra log directory. Let me know if that works for you!

Regarding the MPS trainer, this actually depends on [Pytorch Lightning](https://lightning.ai/docs/pytorch/stable/), so I can't really tell when it will be fully supported...
